artifactory/commands/gradle/gradle.go

```diff
@@ -141,6 +149,95 @@ func (gc *GradleCommand) unmarshalDeployableArtifacts(filesPath string) error {
 	return nil
 }
 
+// runWithFlexPack executes Gradle using FlexPack for dependency resolution and build info collection
+func (gc *GradleCommand) runWithFlexPack() error {
+	log.Debug("Gradle FlexPack implementation activated")
+
+	// Get Gradle executable path
+	gradleExecPath, err := getGradleExecutablePath()
+	if err != nil {
+		return fmt.Errorf("failed to find Gradle executable: %w", err)
+	}
+
+	// Execute Gradle command directly (no JFrog Gradle plugin)
+	cmd := exec.Command(gradleExecPath, gc.tasks...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		log.Error("Failed to execute Gradle command: " + err.Error())
+		return errorutils.CheckError(err)
+	}
+
+	// Check if build info collection is requested
+	if gc.configuration != nil {
+		isCollect, err := gc.configuration.IsCollectBuildInfo()
+		if err != nil {
+			return err
+		}
+		if isCollect {
+			log.Info("Collecting build info for executed command...")
+
+			buildName, err := gc.configuration.GetBuildName()
+			if err != nil {
+				return err
+			}
+			buildNumber, err := gc.configuration.GetBuildNumber()
+			if err != nil {
+				return err
+			}
```

Comment:
nit pick: can this refactored into separate function like getBuildNameAndNumber()

```
func (gc *GradleCommand) getBuildNameAndNumber() (string, string, error) {

}
```